### DISCUSSION TITLE
[RFC] Tooltip support `arrow` prop

### DISCUSSION THREAD

**Discussion by MadCcc** [ant-design/ant-design#40099] [RFCs - archive]
## Summary

Tooltip support new prop `arrow` to control if arrow is displayed and its position.

## Activation

In [rc-tooltip](https://github.com/react-component/tooltip), we provide `showArrow` to control if arrow is displayed. And this prop could be used in Tooltip and other components based on Tooltip in antd though it is not written in API Table. However, `showArrow` could not affect the gap between target element and the popup element, so usually we have to use with another prop not in API Table - `align`.

So finally we want to add a new prop like Dropdown to solve this problem.

## API

**Tooltip**

| Property | Description | Type | Default | 
| --- | --- | --- | --- | 
| arrow | Control arrow element | `boolean \| { pointAtCenter: boolean }` | true | 

## Basic Example

```tsx
import React from 'react';
import { Tooltip, Button } from 'antd';

export default () => (
  <Tooltip title="This tooltip has no arrow" arrow={false}>
    <Button>button</Button>
  </Tooltip>
)
```

## Detailed Design

Add classname `ant-tooltip-show-arrow` when `arrow` is not falsy, and adjust gap between target element and the popup element if arrow is hidden. `arrowPointAtCenter` should be merged into `arrow={{ pointAtCenter: true }}`

This feature should be adpated to Tooltip, Popover, Popconfirm.

## How we teach this

In **feature** branch:

1. Add `arrow` in API Table
2. Deprecate `arrowPointAtCenter` and give warning.

**Comment by crazyair** [comment]
这个目前是用 css 强制把 arrow 隐藏的

**Comment by kiner-tang** [comment]
新增了`arrow`属性后，`showArrow`属性就应该标记为过期属性了吧

**Reply by MadCcc** [reply]
rc-tooltip 可以先不动，主要是 antd 新增

**Reply by kiner-tang** [reply]
好的

**Comment by kiner-tang** [comment]
<img width="898" alt="image" src="https://user-images.githubusercontent.com/10286961/211548593-e9b1a387-e4f4-46f3-a286-cec0d58e8416.png">
<img width="898" alt="image" src="https://user-images.githubusercontent.com/10286961/211548929-e4ebb557-8fd3-475b-876d-c1111e1dfc0f.png">

我们目前`rc-tooltip`组件上好像没有这个`API`呀

**Reply by afc163** [reply]
antd 有的：https://ant.design/components/tooltip-cn

**Reply by MadCcc** [reply]
在 antd 里改

**Reply by kiner-tang** [reply]
好的
